mos-platform/pce/libpce/include/pce/vdc.h

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2023 Adrian "asie" Siekierka
+ *
+ * Licensed under the Apache License, Version 2.0 with LLVM Exceptions,
+ * See https://github.com/llvm-mos/llvm-mos-sdk/blob/main/LICENSE for license
+ * information.
+ */
+
+#ifndef _PCE_VDC_H_
+#define _PCE_VDC_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** \file vdc.h
+ * Functionality related to the VDC.
+ */
+
+#define VDC_SPRITE_COLOR(n)     ((n))
+#define VDC_SPRITE_COLOR_MASK   (0xF)
+#define VDC_SPRITE_BG           0
+#define VDC_SPRITE_FG           (0x1 << 7)
+#define VDC_SPRITE_WIDTH_16     0
+#define VDC_SPRITE_WIDTH_32     (0x1 << 8)
+#define VDC_SPRITE_WIDTH_MASK   (0x1 << 8)
+#define VDC_SPRITE_HEIGHT_16    0
+#define VDC_SPRITE_HEIGHT_32    (0x1 << 12)
+#define VDC_SPRITE_HEIGHT_64    (0x3 << 12)
+#define VDC_SPRITE_HEIGHT_MASK  (0x3 << 12)
+#define VDC_SPRITE_FLIP_X       (0x1 << 11)
+#define VDC_SPRITE_FLIP_Y       (0x1 << 15)
+
+typedef struct {
+    uint16_t y;
+    uint16_t x;
+    uint16_t pattern;
+    uint16_t attr;
+} vdc_sprite_t;
+
+/**
+ * @brief Read a value from the VDC port.
+ * 
+ * @param index Register to read.
+ * @return uint16_t Value read.
+ */
+uint16_t pce_vdc_peek(uint8_t index);
+
+/**
+ * @brief Write a value to the VDC port.
+ * 
+ * @param index Register to write.
+ * @param data Value to write.
+ */
+void pce_vdc_poke(uint8_t index, uint16_t data);
+
+/**
+ * @brief Set the VDC's copy method to copy every word.
+ */
+void pce_vdc_set_copy_word(void);
+/**
+ * @brief Set the VDC's copy method to copy every row.
+ * Alias for @ref pce_vdc_set_copy_word .
+ */
+#define pce_vdc_set_copy_row() pce_vdc_set_copy_word()
```

Comment:
Actually I can never keep the semantics of C inline in my head; I've no idea whether it'll actually de-dup static inline versions of these. I've been in the habit of just making these kinds of things real functions in the .c file; it's pretty reasonable to assume that if you want fast code on llvm-mos you're doing LTO.

Reply:
Admittedly, I've been in the habit of doing weird things like this because most previous platforms I worked with either lacked LTO or had buggy LTO.

Do you think it's a good idea to just standardize on non-static/non-inline functions across the library, then?